cpp/src/fil/infer.cu

```diff
@@ -167,9 +166,10 @@ struct tree_aggregator_t {
 
   /** 
   num_classes is used for other template parameters */
-  __device__ __forceinline__ tree_aggregator_t(int num_classes,
-                                               void* shared_workspace, size_t)
-    : tmp_storage(shared_workspace) {}
+  __device__ __forceinline__ tree_aggregator_t(predict_params params,
+                                               void* accumulate_workspace,
+                                               void* finalize_workspace)
```

Comment:
Any reason two shared memory pointers are now needed (instead of one previously)?

Reply:
It replaces an integer size parameter, it's not like the signature grew.
The math seems more obvious to me this way - fewer classes need to consider the footprint and placement of input columns, which is decided in the kernel body, as opposed to `tree_aggregator_t` constructors.

Reply:
This is fine.